### DISCUSSION TITLE
addcol-expr and rename?

### DISCUSSION THREAD

**Discussion by halloleo** [saulpw/visidata#1144] [Q&A]
I use `addcol-expr` a lot, but one thing I notced is the the default names are always oretty unwieldy. E.g. when I do:

    ='' if visibleCols[1].getValue(row)==visibleCols[2].getValue(row) else visibleCols[2].getValue(row)

the column name is `'' if visibleCols[`...

Of course I then change the name with `^`, but I wanted to ask, whether there is a way to add a column _and_ give it a name in one operation...

***

PS: I am just starting to use `record` and `play` which is extermely promising!

**Comment by saulpw** [comment]
Hey @halloleo, I think this would be a useful feature too, and I had an idea for this syntax (not implemented yet): `<identifier>=<expr>` which would use `<expr>` for the column expr but `<identifier>` for the column name.  What do you think about this idea?

**Reply by halloleo** [reply]
Great you think this would be an improvement. However, I don't understand the syntax you suggest. "=" is already the reserved key binding for `addcol-expr` as it is used now, right?

**Reply by saulpw** [reply]
Sorry, I had angle brackets in the comment which were being turned into HTML.  Fixed now in the comment above.  The idea is that you'd type `=foo=n+1` and the first `=` would invoke `addcol-expr`, and then the input would be `foo=n+1` which would name the column `foo` while giving it the expression `n+1`.

**Reply by halloleo** [reply]
Got it. Sounds great! 

I think the two equal signs are reasonably intuitive _and_ it is backward compatible with the current syntax (where you have only one equal sign).

**Reply by jazzabeanie** [reply]
You could make 'new column expr=' editable and if it is changed it sets the name for the column. Eg, I press `=`, then I see
`new column expr=`, then I backspace and change it to `new_column='some_expr'`. This then creates a new column with the name 'new_column'.

**Reply by jcmkk3** [reply]
> You could make 'new column expr=' editable and if it is changed it sets the name for the column. Eg, I press `=`, then I see `new column expr=`, then I backspace and change it to `new_column='some_expr'`. This then creates a new column with the name 'new_column'.

This is kind of how PowerBI does it and it seems pretty intuitive. If you click to create a new measure, then you get this:
![image](https://user-images.githubusercontent.com/45919695/157064855-e3512525-a578-449b-aaa5-6d6441ff802f.png)

You could decide to just leave the generic name or rename it to make more sense.
![image](https://user-images.githubusercontent.com/45919695/157064968-f6260c19-3253-4a51-9b17-38382df2a3e9.png)

If you already had a measure named `Measure`, then it would have offered the default measure name as `Measure 2` instead.
